src/main/java/nl/workingtalent/wtacademy/reservation/ReservationController.java

```diff
@@ -1,11 +1,140 @@
 package nl.workingtalent.wtacademy.reservation;
 
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import nl.workingtalent.wtacademy.book.Book;
+import nl.workingtalent.wtacademy.book.BookService;
+import nl.workingtalent.wtacademy.dto.ResponseDto;
+import nl.workingtalent.wtacademy.user.User;
+import nl.workingtalent.wtacademy.user.UserService;
+
 @RestController
+@CrossOrigin(maxAge = 3600)
 public class ReservationController {
 
 	@Autowired
 	private ReservationService service;
+
+	@Autowired
+	private BookService bookService;
+
+	@Autowired
+	private UserService userService;
+
+	// READ
+	@RequestMapping("reservation/all")
+	public ResponseDto findAllReservations() {
+		List<Reservation> reservations = service.findAllReservations();
+
+		if (reservations.isEmpty()) {
+			return new ResponseDto(false, null, null, "No reservations found.");
+		}
+
+		Stream<ReadReservationDto> readReservationDtoStream = getReservations(reservations);
+		return new ResponseDto(true, readReservationDtoStream, null,
+				reservations.size() + (reservations.size() < 2 ? " reservation " : " reservations ") + "found.");
+	}
```

Comment:
Dit kan misschien ook omgezet worden naar pages. Dan zou je waarschijnlijk sorteren zodat de nieuwste reservation requests bovenaan staan. Werkt nu goed, kan later worden toegevoegd.

Reply:
Ik zal het in de backlog zetten, zodat we het niet vergeten.